resolvers/node/package_json.rs

```diff
@@ -14,12 +14,31 @@ use url::Url;
 use crate::errors::ClosestPkgJsonError;
 use crate::errors::PackageJsonLoadError;
 
-#[allow(clippy::disallowed_types)]
-pub type PackageJsonCacheRc = crate::sync::MaybeArc<
-  dyn deno_package_json::PackageJsonCache
+pub trait NodePackageJsonCache:
+  deno_package_json::PackageJsonCache
+  + crate::sync::MaybeSend
+  + crate::sync::MaybeSync
+{
+  fn as_deno_package_json_cache(
+    &self,
+  ) -> &dyn deno_package_json::PackageJsonCache;
+}
+
+impl<T> NodePackageJsonCache for T
+where
+  T: deno_package_json::PackageJsonCache
     + crate::sync::MaybeSend
     + crate::sync::MaybeSync,
->;
+{
+  fn as_deno_package_json_cache(
+    &self,
+  ) -> &dyn deno_package_json::PackageJsonCache {
+    self
+  }
+}
```

Comment:
I shouldn't have added Debug to PackageJsonCache in deno_package_json and instead just done this with Debug.

Reply:
Seems like https://github.com/denoland/deno/issues/27829 is related